crates/io/src/store.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{Format, Readable, Result, Writeable};
+use object_store::{ObjectStore, PutResult, path::Path};
+use stac::Href;
+use std::sync::Arc;
+
+/// Parses an href into a [StacStore] and a [Path].
+pub fn parse_href(href: impl AsRef<Href>) -> Result<(StacStore, Path)> {
+    parse_href_opts(href, [] as [(&str, &str); 0])
+}
+
+/// Parses an href and options into [StacStore] and a [Path].
+///
+/// Relative string hrefs are made absolute `file://` hrefs relative to the current directory.`
+pub fn parse_href_opts<I, K, V>(href: impl AsRef<Href>, options: I) -> Result<(StacStore, Path)>
+where
+    I: IntoIterator<Item = (K, V)>,
+    K: AsRef<str>,
+    V: Into<String>,
+{
+    let (store, path) = match href.as_ref() {
+        Href::Url(url) => object_store::parse_url_opts(url, options)?,
+        Href::String(s) => {
+            if s.starts_with("/") {
+                object_store::parse_url_opts(&format!("file://{s}").parse()?, options)?
+            } else {
+                let s = std::env::current_dir()?.join(s);
+                object_store::parse_url_opts(&format!("file://{}", s.display()).parse()?, options)?
+            }
+        }
+    };
+    Ok((store.into(), path))
+}
+
+/// Reads STAC from an [ObjectStore].
+#[derive(Debug)]
+pub struct StacStore(Arc<dyn ObjectStore>);
```

Comment:
This is just a wrapper around the underlying `ObjectStore`... I don't really see the value in that. You're still not allowing users to use anything other than an `ObjectStore`.

I was thinking more of defining your own trait like https://docs.rs/parquet/latest/parquet/arrow/async_reader/trait.AsyncFileReader.html so that users wouldn't be forced into using `object_store`

Reply:
I guess the value that I see is in providing the format translations for you? I was apeing https://docs.rs/parquet/latest/parquet/arrow/async_reader/struct.ParquetObjectReader.html

Reply:
The point of `ParquetObjectReader` is not that it simply wraps an `ObjectStore`; the point is that [it implements `AsyncFileReader`](https://docs.rs/parquet/latest/parquet/arrow/async_reader/struct.ParquetObjectReader.html#impl-AsyncFileReader-for-ParquetObjectReader).

Reply:
If you don't have a trait here, I don't really see the point of the change in this PR. 

That said, it's fine if you don't want the complexity at this stage of allowing users to use something other than `object_store`

Reply:
Yeah, I think I need to think about it more to see how/when I'll need a trait. Right now I'm leaning _hard_ into `object_store` for everything `async`, and I don't see _yet_ the need to support other backends, since generally STAC is get-and-put (direct byte access isn't really in scope for the repo, as of yet).

The wrapper struct is a nicer API than what I had, where I dangled a bunch of crappy methods off of `Format`, so I'm still calling it a win :-P.